### PR TITLE
enh(core): Use SWC as a loader in order to boost build speed

### PR DESCRIPTION
**Pull request by Thebarda**
## Description

This replaces babel-loader/ts-loader with swc-loader in order to to simplify our Webpack configuration and speed up our build.

Before (development mode/production mode):

Screenshot 2022-05-18 at 11 23 56
Screenshot 2022-05-18 at 11 22 29

After:

Screenshot 2022-05-18 at 10 52 57
Screenshot 2022-05-18 at 10 52 22

NOTE: Builds were launched using 4 vCPU and 8 MB of memory so it may differ with your build time.
NOTE 2: Oh and it removes Redux from our codebase.

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

<h2> How this pull request can be tested ? </h2>

- Run a production build using `npm run build:prod`
- Run a development build using `npm run build:dev`
- Run tests `npm t`
- -> Those commands should work as before

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/front_src/src/App/index.tsx [hunk 1]
```diff
@@ -64,27 +60,25 @@ const App = (): JSX.Element => {
   const min = hasMinArgument();
 
   return (
-    <ReduxProvider store={store}>
-      <Suspense fallback={<PageLoader />}>
-        <div className={classes.wrapper}>
+    <Suspense fallback={<PageLoader />}>
+      <div className={classes.wrapper}>
+        {not(min) && (
```
Comment (bdauria) [inline www/front_src/src/App/index.tsx:65]: Do we still need the min parameter?

File: www/front_src/src/helpers/dynamicImport.js [hunk 2]
```diff
@@ -64,9 +62,9 @@ export const dynamicImport = (basename, parameters) =>
 
     try {
       // dynamically import css if external component needs one
```
Comment (bdauria) [inline www/front_src/src/helpers/dynamicImport.js:64]: We can get rid of this now

File: www/front_src/src/App/index.tsx (hunk 1 above)
Comment (Thebarda) [reply www/front_src/src/App/index.tsx:65]: Yep, this is required for the autologin

File: www/front_src/src/helpers/dynamicImport.js (hunk 2 above)
Comment (Thebarda) [reply www/front_src/src/helpers/dynamicImport.js:64]: We will get rid of it in the Module Federation PR. It's not related to that PR

**Pull request closed without merging**